datastore/longhorn.go

```diff
@@ -585,29 +585,73 @@ func checkEngine(engine *longhorn.Engine) error {
 	return nil
 }
 
-// PickVolumeCurrentEngine pick the current Engine from the Engine list of a volume with the given namespace
-func (s *DataStore) PickVolumeCurrentEngine(v *longhorn.Volume, es map[string]*longhorn.Engine) (*longhorn.Engine, error) {
-	if len(es) == 0 {
-		return nil, nil
+// GetCurrentEngineAndExtras pick the current Engine and extra Engines from the Engine list of a volume with the given namespace
+func GetCurrentEngineAndExtras(v *longhorn.Volume, es map[string]*longhorn.Engine) (currentEngine *longhorn.Engine, extras []*longhorn.Engine, err error) {
+	if len(es) == 1 {
+		for _, e := range es {
+			return e, nil, nil
+		}
 	}
+	for _, e := range es {
+		if e.Spec.Active {
```

Comment:
What happens if there are multiple active engines should we overwrite? Or is there another condition we can use in that case? i.e first come first server?

Reply:
Right now it's first come first served. I am hesitating if we need to error out when there are multiple active engines. Same below.

Reply:
If possible, in general we should have debug/warn log, because it's better for debugging.

Reply:
OK... Let's directly error out for now. When multiple active engines are found, continuing the execution is weird...